libraries/WiFi/src/WiFiClass.cpp

```diff
@@ -67,30 +67,16 @@ void WiFiClass::mode(WiFiMode_t m) {
     }
 }
 
-
-/*  Start WiFi connection for OPEN networks
-
-    param ssid: Pointer to the SSID string.
-*/
-int WiFiClass::begin(const char* ssid) {
-    return begin(ssid, nullptr);
-}
-
-int WiFiClass::beginBSSID(const char* ssid, const uint8_t *bssid) {
-    return begin(ssid, nullptr, bssid);
-}
-
-/*  Start WiFi connection with passphrase
-    the most secure supported mode will be automatically selected
-
-    param ssid: Pointer to the SSID string.
-    param passphrase: Passphrase. Valid characters in a passphrase
-          must be between ASCII 32-126 (decimal).
-*/
-int WiFiClass::begin(const char* ssid, const char *passphrase, const uint8_t *bssid) {
+bool WiFiClass::_beginInternal(const char* ssid, const char *passphrase, const uint8_t *bssid) {
     // Simple ESP8266 compatibility hack
     if (_modeESP == WIFI_AP) {
-        return beginAP(ssid, passphrase);
+        // When beginAP was a success, it returns WL_CONNECTED. Therefore we return false as no error when that happens.
+        int status = beginAP(ssid, passphrase);
+        if (status != WL_CONNECTED) {
+            return true;
```

Comment:
Err...are you a Linux or bash user?  0(false) for success, !=0(true) for fail is fine for the shell but backwards for most C I've ever seen, and opposite of normal in this code, too.  (i.e. `if !_wifi.begin()) return WL_IDLE_STATUS`).

To keep things consistent, could you please swap the logic so `::_beginInternal` returns `true` on success and `false` on failure?

Reply:
Woops good point, I will change that soon! 

I usually chose to return _above_ 0 as error, because you usually can have many different errors. So you can identify them if needed.
Of course theoretically you can have many different successes as well, but these don't matter too much 😄 

This is the reason for me returning an int, not a bool and above 0 as error. _But now I mixed them up_! It does not matter that much anyway.
However, for consistency I will change it. Inconsistent code is annoying :)

For the future, I will make sure I will double check consistency. Arduino style is not something I come from ^^